node/overseer/overseer-gen/src/lib.rs

```diff
@@ -196,13 +196,15 @@ pub struct SignalsReceived(Arc<AtomicUsize>);
 impl SignalsReceived {
 	/// Load the current value of received signals.
 	pub fn load(&self) -> usize {
-		// off by a few is ok
-		self.0.load(atomic::Ordering::Relaxed)
+		// It's imperative that we prevent reading a stale value from memory because of reordering.
+		// Memory barrier to ensure that no reads or writes in the current thread before this load are reordered.
+		// All writes in other threads using release semantics become visible to the current thread.
+		self.0.load(atomic::Ordering::Acquire)
```

Comment:
Just in case, there is no difference on x86_64 between Relaxed/Acquire/SeqCst for loads. They are translated just to a single `mov` operation without locking as everything else is guaranteed by the ISA.